### PR TITLE
New mention object and small cleanups

### PR DESCRIPTION
**Pull request by davfsa**
### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

File: hikari/impl/bot.py [hunk 1]
```diff
@@ -219,26 +219,22 @@ class BotApp(traits.BotAware, event_dispatcher.EventDispatcher):
...
-        "incremental": True,
-        "loggers": {
-            "hikari.gateway": {
-                "level": "DEBUG"
-            },
-            "hikari.ratelimits": {
-                "level": "TRACE_HIKARI"
+    bot = hikari.Bot(
+        token=os.environ["BOT_TOKEN"],
+        intents=hikari.Intents.ALL,
```
[the remark this answers is not in the document]
Comment (davfsa) [reply hikari/impl/bot.py:227]: Will remove that

File: hikari/messages.py [hunk 2]
```diff
@@ -209,6 +213,127 @@ class MessageActivity:
...
+    @property
+    def crossposted_channels_ids(self) -> typing.Sequence[snowflakes.Snowflake]:
+        return list(self.crossposted_channels.keys())
+
+    @property
+    def notifies_user_ids(self) -> typing.Sequence[snowflakes.Snowflake]:
+        return list(self.notifies_users.keys())
+
+    @property
+    def notifies_members(self) -> typing.Mapping[snowflakes.Snowflake, guilds.Member]:
```
[the remark this answers is not in the document]
Comment (davfsa) [reply hikari/messages.py:250]: its to keep consistent with all the others attrs

[the remark this answers is not in the document]
Comment (FasterSpeeding) [reply hikari/messages.py:250]: consistent with what other attrs?

**Review by FasterSpeeding — DISMISSED** [review]

File: hikari/messages.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (davfsa) [reply hikari/messages.py:250]: Hmm, good point, will change

[the remark this answers is not in the document]
Comment (davfsa) [reply hikari/messages.py]: Fixed :+1:

File: hikari/impl/entity_factory.py [hunk 3]
```diff
@@ -1580,8 +1590,14 @@ def deserialize_partial_message(self, payload: data_binding.JSONObject) -> messa
             message_reference=message_reference,
             flags=message_models.MessageFlag(payload["flags"]) if "flags" in payload else undefined.UNDEFINED,
             nonce=payload["nonce"] if "nonce" in payload else undefined.UNDEFINED,
+            # We initialize these next.
+            mentions=NotImplemented,
         )
 
+        message.mentions = self._deserialize_mentions(payload, message)
```
Comment (FasterSpeeding) [inline hikari/impl/entity_factory.py:1597]: How would we tell the difference between the mentions being changed to no mentions vs the mentions not being changed (left unset and therefor not provided in the payload for message updates like embed specific ones)

Comment (davfsa) [reply hikari/impl/entity_factory.py:1597]: Forgot to implement, will change

Comment (davfsa) [reply hikari/impl/entity_factory.py]: Implemented :+1:

**Review by FasterSpeeding — APPROVED** [review]

**Pull request merged**